modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/AttachmentObjectFieldBusinessType.java

```diff
@@ -275,6 +293,56 @@ private void _validateObjectFieldSettingMaximumFileSize(
 		}
 	}
 
+	private void _validateObjectFieldSettingShowFiles(
+			String objectFieldName, String showFilesInDocumentsAndMedia,
+			String storageFolder)
+		throws PortalException {
+
+		if (Validator.isNull(showFilesInDocumentsAndMedia)) {
+			throw new ObjectFieldSettingValueException.MissingRequiredValues(
+				objectFieldName,
+				Collections.singleton("showFilesInDocumentsAndMedia"));
+		}
+		else if (StringUtil.equalsIgnoreCase(
+					showFilesInDocumentsAndMedia, StringPool.TRUE)) {
+
+			_validateObjectFieldSettingStorageFolder(
+				true, objectFieldName, storageFolder);
+		}
+		else if (StringUtil.equalsIgnoreCase(
+					showFilesInDocumentsAndMedia, StringPool.FALSE)) {
+
+			_validateObjectFieldSettingStorageFolder(
+				false, objectFieldName, storageFolder);
+		}
+		else {
+			throw new ObjectFieldSettingValueException.InvalidValue(
+				objectFieldName, "showFilesInDocumentsAndMedia",
+				showFilesInDocumentsAndMedia);
+		}
+	}
+
+	private void _validateObjectFieldSettingStorageFolder(
+			boolean allowed, String objectFieldName, String storageFolder)
+		throws PortalException {
+
+		if (!allowed && Validator.isNotNull(storageFolder)) {
+			throw new ObjectFieldSettingNameException.NotAllowedNames(
+				objectFieldName, Collections.singleton("storageFolder"));
```

Comment:
Does it make sense to do the same thing with `showFilesInDocumentsAndMedia` and  `storageFolder` in case the `fileSource` is `documentsAndMedia`?

Reply:
Yeah... I think it does. I'll do that too. Thanks!